docs/sdk-configuration.md

```diff
@@ -0,0 +1,492 @@
+with their internal infrastructure and as such have some familiarity with tracing at least as the
+SDK presents it.
+
+- OpenTelemetry maintainers. Write the SDK code.
+
+When making decisions, especially about complexity, we always prioritize the application developers,
+then framework developers, then maintainers. This is because we expect those with less domain
+knowledge about tracing to require a simpler experience than those with more. There is also more
+bang-for-the-buck by making the end-user experience as streamlined as possible since we expect there
+to be much more of them than other audiences.
+
+## Goals and non-goals
+
+### Goals
+
+- Provide a single entrypoint to configuring the SDK. For end-users, less familiar with the SDK, we
+want to have everything together to provide discoverability and simpler end-user code.
+
+- Fit well with common Java idioms such as dependency injection, or common frameworks like Spring.
+
+- Reduce the chance of gotchas or configuration mishaps.
+
+- Aim for optimal performance for the most common use case.
+
+### Non-goals
+
+- Provide the best possible experience for custom SDKs. Generally the burden of the experience for
+custom SDKs can fall on their authors, we optimize for our standard usage, the full SDK. Any
+reference to "the SDK" in this document refers to the full SDK with all signals.
+
+- Make sure everything is auto-configurable. This is out of the scope of the SDK, and instead is
+left to auto-configuration layers, which are also described below but not as part of the core SDK. 
+In partiular, SignalProvider SPIs which currently exist are proposed to be removed from the SDK. The 
+SDK may provide an autoconfiguration extension as an option which is not internal to the main SDK 
+components.
+
+## Configuring an instance of the SDK
+
+The SDK consists of configuration options for all the signals it supports. Users all have different
+requirements for how they use the SDK, for example they may use different exporters depending on
+their backend. Because we cannot guess the configuration the user needs, we expect that the SDK must
+be configured by the user before it can be used.
+
+Goals for configuring the SDK are
+
+- Discoverability of options
+- Ease of use by end users, e.g., less complicated code required
+- Avoid requiring duplicate configuration, which can lead to errors or confusion
+- Provide good defaults where possible
+
+In Java, the builder pattern is common for configuring instances. Let's look at what that may look
+like. The simplest configuration will be when a user wants to get a default experience, exporting
+with a specific exporter to an endpoint.
+
+```java
+OpenTelemetrySdk openTelemetry = OpenTelemetrySdk.builder()
+  .addSpanExporter(OtlpGrpcSpanExporter.builder()
+    .setEndpoint("spans-service:4317")
+    .build())
+  .addMetricExposition(PrometheusExposition.getDefault())
```

Comment:
This example mixes two signals, which is good. The question is: are we likely to end up with an unwieldy builder if it contains the configuration for multiple signals, especially when we (eventually) add logs in here as well.

Would we be better off (as I think Bogdan has suggested) requiring the SDK components for each signal to be configured indivually, then have the `OpenTelemetrySdk` builder only compose the other components? I can see pros and cons.